tests/contrib/create_failing_test_for_compare_ids_test.bzl

```diff
@@ -0,0 +1,103 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Implementation of create_failing_test_for_compare_ids_test rule
+def _create_failing_test_for_compare_ids_test_impl(ctx):
+    test_code = """ '
+load("//:compare_ids_test.bzl", "compare_ids_test")
+
+compare_ids_test(
+    name = "test",
+    id = {id},
+    tars = {tars},
+)
+'
+    """.format(
+        id = repr(ctx.attr.id),
+        # When linked, the tars will be placed into the base folder, and renamed 0.tar, 1.tar, ... etc
+        tars = repr([str(i) + ".tar" for i in range(len(ctx.attr.tars))]),
+    )
+
+    tar_files = []
+    for tar in ctx.attr.tars:
+        tar_files += list(tar.files)
```

Comment:
Is there a way that we can get the single file from tar?

Reply:
Because the image targets aren't actually files, bazel doesn't know how many files they 'output'. Normally there should be only one but, this method allows using one target that outputs multiple tarballs. If you prefer, I could change it to take just the first tarball in the file list, and make sure the list is of length 1. Thoughts?